xarray/core/formatting.py

```diff
@@ -145,15 +145,50 @@ def format_array_flat(items_ndarray, max_width):
     return pprint_str
 
 
+def _get_level_values(index, level_num, max_size=50):
+    """Similar to `pd.MultiIndex.get_level_values` but
+    here truncated to `max_size`.
+    """
+    unique = index.levels[level_num]
+    labels = index.labels[level_num]
+    size = min(max_size, labels.size)
+    filled = pd.core.algorithms.take_1d(unique.values, labels[:size],
```

Comment:
Yes, that's a good point. It's definitely more useful to display the particular index levels that are actually used at those points.

In that case I would use `unique[labels[:max_size]]`. It's better to avoid the NumPy methods like `np.take` on pandas.Index objects because they don't always always preserve dtypes properly.